### DISCUSSION TITLE
Is the connection pool shared across multiple instances?

### DISCUSSION THREAD

**Discussion by yiweig** [pgbouncer/pgbouncer#1128] [Q&A] [answered]
Maybe a dumb question, but I'd like to get more information on PGBouncer behavior for our use case to make sure we're using it properly. 

Context: we're running PGBouncer as a sidecar in GCP Cloud Run. Our main app container is a Django app, and we're connecting to Cloud SQL Postgres. If it matters, we're using the Bitnami PGBouncer image. 

When Cloud Run scales up our containers, the number of PGBouncer instances also scales up accordingly. Our app is single tenant, so all of our Django instances connects through PGBouncer using the same username and password.

My question: I understand that the docs say that `default_pool_size` dictates "How many server connections to allow per user/database pair". But is that per PGBouncer instance? Or is that globally? 

That is, when our service scales up to 5 containers (ie 5 PGBouncer instances), do we get a total connection pool of `5 x 20 = 100`? Or will it remain a pool of `20` connections?

Thanks!

**Comment by JelteF** [comment, marked as answer]
All limits are per PgBouncer instance. So indeed with 5 instances you would get `5 x 20 = 100`.

**Reply by yiweig** [reply]
Thanks @JelteF! I figured it was like this, but re-assuring to hear it from you. Really appreciate all the work you and everyone have put into PGBouncer! It's great :)

**Comment by emily-ginsburg** [comment]
Hi! Following up here. I have a similar set up - setting up PgBouncer on my Django/Postgres/Cloud Run app - and now have been seeing some talk online about the option to run PgBouncer as one shared system of connection pooling used across my multiple Cloud Run instances. Based on this response, it's unclear to me if that's feasible. Is this type of system an option? And if so, is it recommended/documented?
